.github/workflows/build_and_test.yml

```diff
@@ -251,9 +264,20 @@ jobs:
         bash miniconda.sh -b -p $HOME/miniconda
     # Run the tests.
     - name: Run tests
+      env: ${{ fromJSON(needs.configure-jobs.outputs.envs) }}
       run: |
+        # TODO(SPARK-36361): Install coverage in Python 3.9 and PyPy 3 in the base image
+        python3.9 -m pip install coverage
+        pypy3 -m pip install coverage
         export PATH=$PATH:$HOME/miniconda/bin
         ./dev/run-tests --parallelism 1 --modules "$MODULES_TO_TEST"
+    - name: Upload coverage to Codecov
+      if: needs.configure-jobs.outputs.type == 'pyspark-coverage-scheduled'
+      uses: codecov/codecov-action@v1
```

Comment:
qq: Just not sure what's the diff between v1 and v2, looks like v2 is the recommand version in codecov-action.

[1] https://github.com/codecov/codecov-action#usage